force-app/main/default/classes/RD2_StatusAutomationService.cls

```diff
@@ -74,51 +74,98 @@ public with sharing class RD2_StatusAutomationService {
+        automationSettings.StatusAutomationClosedValue__c = closedStatus;
+
+        return automationSettings;
+    }
+    public void saveSetting(npe03__Recurring_Donations_Settings__c automationSettings) {
         validateUpdatePermission();
-        npe03__Recurring_Donations_Settings__c settings = 
-            UTIL_CustomSettingsFacade.getRecurringDonationsSettings();
-        
-        settings.StatusAutomationDaysForLapsed__c = daysForLapsed;
-        settings.StatusAutomationDaysForClosed__c = daysForClosed;
-        settings.StatusAutomationLapsedValue__c = lapsedStatus;
-        settings.StatusAutomationClosedValue__c = closedStatus;
 
         if (!Test.isRunningTest()) {
-            update settings;
+            update automationSettings;
         }
     }
 
-    public String validateSetting(Integer daysForLapsed, Integer daysForClosed, String lapsedStatus, String closedStatus) {
-        if (daysForLapsed != null && daysForClosed != null) {
-            if (daysForLapsed >= daysForClosed) {
+    public Boolean isSettingValid() {
+        return String.isBlank(validateSetting(UTIL_CustomSettingsFacade.getRecurringDonationsSettings()));
+    }
+
+    public String validateSetting(npe03__Recurring_Donations_Settings__c settings) {
+        if (settings.StatusAutomationDaysForLapsed__c != null
+            && settings.StatusAutomationDaysForClosed__c != null) {
+            if (settings.StatusAutomationDaysForLapsed__c >= settings.StatusAutomationDaysForClosed__c) {
                 return System.label.RD2_StatusAutomationInvalidNumberOfDays;
             }
         }
 
-        if (daysForLapsed != null) {
-            if (statusMapper.getState(lapsedStatus) != RD2_Constants.STATUS_LAPSED) {
+        if (settings.StatusAutomationDaysForLapsed__c != null) {
+            if (statusMapper.getState(settings.StatusAutomationLapsedValue__c) != RD2_Constants.STATUS_LAPSED) {
                 return System.label.RD2_StatusAutomationInvalidLapsedStatus;
             }
 
-            if (daysForLapsed < 0) {
+            if (settings.StatusAutomationDaysForLapsed__c < 0) {
                 return System.label.RD2_StatusAutomationNegativeNumberOfDays;
             }
         }
 
-        if (daysForClosed != null) {
-            if (statusMapper.getState(closedStatus) != RD2_Constants.STATUS_Closed) {
+        if (settings.StatusAutomationDaysForClosed__c != null) {
+            if (statusMapper.getState(settings.StatusAutomationClosedValue__c) != RD2_Constants.STATUS_Closed) {
                 return System.label.RD2_StatusAutomationInvalidClosedStatus;
             }
 
-            if (daysForClosed < 0) {
+            if (settings.StatusAutomationDaysForClosed__c < 0) {
                 return System.label.RD2_StatusAutomationNegativeNumberOfDays;
             }
         }
 
         return null;
     }
 
+    public Boolean shouldMarkRDAsLapsed(RD2_RecurringDonation rdRecord, RD2_ScheduleService scheduleService) {
```

Comment:
Not a deal breaker, but I could see these two methods being combined into one

Reply:
Yup, I'd suggest extracting this:

    private Boolean isEligibleForStatusUpdate(RD2_RecurringDonation rdRecord) {
        if (!new RD2_StatusAutomationService().isSettingValid()
            || !rd2Settings.isStatusAutomationForClosedEnabled()
            || rdRecord.isClosed()
            || rdRecord.isPaused()
            || !rdRecord.hasSchedule()
            || rdRecord.isElevateRecord()
        ) {
            return false;
        }
        return true;
    }